### DISCUSSION TITLE
Scanning non-HTTP protocols

### DISCUSSION THREAD

**Discussion by PurushottamanR** [projectdiscovery/nuclei#605] [Q&A] [answered]
Will you also be adding features to scan for services that work without the web interface for example ftp and other protocols?
For example we want to fuzz an ftp server so it requires us to write templates not as a HTTP request but more of a TCP message with all the related variables.

**Comment by ehsandeep** [comment, marked as answer]
Hey @PurushottamanR, that's what we added in the latest release, check this out how to write a template for those https://nuclei.projectdiscovery.io/templating-guide/protocols/network/ and here are a few [examples](https://nuclei.projectdiscovery.io/template-examples/network/)
